src/api/app/models/user.rb

```diff
@@ -467,8 +467,10 @@ def can_modify_project?(project, ignore_lock = nil)
 
   # FIXME: This should be a policy
   # package is instance of Package
+  # rubocop:disable Metrics/PerceivedComplexity
   def can_modify_package?(package, ignore_lock = nil)
     return false if package.nil? # happens with remote packages easily
+    return false if package.project.scmsync.present? # packages are maintained in scm
```

Comment:
Hm, you just want to avoid storing files to the backend right? This seems to be a bit of overkill, there are lots of database-only things that could/should be writeable.

How about #12277?

Reply:
Are you sure. I can happily add packages and files to a project with `scmsync`

Reply:
Existing objects also do not get removed when adding `scmsync`.

Reply:
no be part of pull request anymore